lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/TaskManager.cs

```diff
@@ -0,0 +1,500 @@
+            {
+                var msg = string.Format(CultureInfo.InvariantCulture, "The task [{0}] doesn't exist.", taskId);
+                Exceptions.Throw(new IMRUSystemException(msg), Logger);
+            }
+
+            var taskInfo = GetTaskInfo(taskId);
+            return taskInfo.Item1.CurrentState;
+        }
+
+        /// <summary>
+        /// Checks if all the tasks are in the state specified. 
+        /// For example, passing TaskState.TaskRunning to check if all the tasks are in TaskRunning state
+        /// </summary>
+        /// <param name="taskState"></param>
+        /// <returns></returns>
+        internal bool AllInTheState(TaskState taskState)
+        {
+            return _tasks.All(t => t.Value.Item1.CurrentState == taskState);
+        }
+
+        /// <summary>
+        /// Submit all the tasks
+        /// Tasks will be submitted after all the tasks are added in the collection and master task exists
+        /// IMRUSystemException will be thrown if not all the tasks are added or if there is no master task
+        /// </summary>
+        internal void SubmitTasks()
+        {
+            if (NumberOfTasks < _totalExpectedTasks || !MasterTaskExists())
+            {
+                string msg = string.Format("Trying to submit tasks but either master task doesn't exist or number of tasks [{0}] is smaller than expected number of tasks [{1}].", NumberOfTasks, _totalExpectedTasks);
+                Exceptions.Throw(new IMRUSystemException(msg), Logger);
+            }
+
+            StartTask(_masterTaskId);
+
+            foreach (var taskId in _tasks.Keys)
+            {
+                if (taskId.Equals(_masterTaskId))
+                {
+                    continue;
+                }
+                StartTask(taskId);
+            }
+        }
+
+        /// <summary>
+        /// Starts a task and then update the task status to submitted.
+        /// </summary>
+        /// <param name="taskId"></param>
+        private void StartTask(string taskId)
+        {
+            var taskInfo = GetTaskInfo(taskId);
+            StartTask(taskId, taskInfo.Item2, taskInfo.Item3);
+            taskInfo.Item1.MoveNext(TaskStateEvent.SubmittedTask);
+        }
+
+        /// <summary>
+        /// Gets GroupCommTaskConfiguration from IGroupCommDriver and merges it with the user partial task configuration.
+        /// Then submits the task with the ActiveContext.
+        /// </summary>
+        /// <param name="taskId"></param>
+        /// <param name="userPartialTaskConf"></param>
+        /// <param name="activeContext"></param>
+        private void StartTask(
+            string taskId,
+            IConfiguration userPartialTaskConf,
+            ITaskSubmittable activeContext)
+        {
+            var groupCommTaskConfiguration = _groupCommDriver.GetGroupCommTaskConfiguration(taskId);
+            var mergedTaskConf = Configurations.Merge(userPartialTaskConf, groupCommTaskConfiguration);
```

Comment:
I'm just curious to find out what is the expected scenario if a user supplies a configuration that conflicts with some configuration option in the `GroupCommTaskConfiguration`.